grammar.js

```diff
@@ -313,6 +313,16 @@ module.exports = grammar({
     [
       $.parenthesized_proc_type,
     ],
+    // TODO: check why these are in conflict
```

Comment:
I may be doing something wrong with my fun def param parsing to necessitate these. What do you think?

Reply:
I believe I have an answer to this. Kind of a long explanation, writing up now.